zebra/rtadv.c

```diff
@@ -1310,10 +1310,11 @@ void ipv6_nd_suppress_ra_set(struct interface *ifp,
 				zif->rtadv.NumFastReXmitsRemain =
 					RTADV_NUM_FAST_REXMITS;
 			}
-
-			rtadv_start_interface_events(zvrf, zif);
 		}
 	}
+
+	/* Always receive RS, no matter RA is disabled or not */
+	rtadv_start_interface_events(zvrf, zif);
```

Comment:
This function does several things - not just reading a socket. I don't think this wants to be unconditionally enabled.

Reply:
fixed